registry/storage/blobstore.go

```diff
@@ -188,10 +349,50 @@ func (bs *blobStatter) Stat(ctx context.Context, dgst digest.Digest) (distributi
 	}, nil
 }
 
+// Clear deletes blob's data with or without its parent directory depending on
+// bs' configuration.
 func (bs *blobStatter) Clear(ctx context.Context, dgst digest.Digest) error {
-	return distribution.ErrUnsupported
+	var (
```

Comment:
That makes sense.  `Clear` is so named because it removes a mapping rather than removes data.